### PR TITLE
Tests: index content unchanged, doc non-deterministic, block_ids with global strategy

### PR DESCRIPTION
**Pull request by redouane-dziri**
Issue #14

### CODE REVIEW

File: tests/test_stochatreat_input_output.py [hunk 1]
```diff
@@ -217,3 +219,21 @@ def test_stochatreat_output_no_null_treats(treatments_dict):
...
+    )
+    treatments = stochatreat(
+        data=data_with_random_index,
+        block_cols= treatments_dict["block_cols"],
+        treats=treatments_dict["n_treatments"],
+        idx_col=treatments_dict["idx_col"],
+        misfit_strategy=misfit_strategy,
+    )
+    asrtmsg = "The output and input indices do not have the same content"
+    assert set(treatments.index) == set(data_with_random_index.index), asrtmsg
```
Comment (RoyalTS) [inline tests/test_stochatreat_input_output.py]: That's a pretty weak test. Could make that much stronger, I think:

1. construct a DataFrame with random but unique index
2. Also assign a unique but random index column
3. pass that to stochatreat in various configs
4. check in the end that the index values are the same
5. check that the correspondence between index and index column are unperturbed

File: tests/test_stochatreat_input_output.py [hunk 2]
```diff
@@ -217,3 +217,88 @@ def test_stochatreat_output_no_null_treats(treatments_dict):
...
+    )
+
+    # align the idx_col of each with the index of the input data
+    idx_cols = pd.concat(
+        [ data_with_random_index[[idx_col]], treatments[[idx_col]] ],
+        axis=1
+    )
+    idx_cols.columns = ["idx_col_data", "idx_col_treatment"]
+
+    asrtmsg = "The output index and idx_col correspondence has been perturbed"
```
Comment (RoyalTS) [inline tests/test_stochatreat_input_output.py]: `assert_msg` for consistency, please!

File: tests/test_stochatreat_input_output.py [hunk 3]
```diff
@@ -217,3 +217,88 @@ def test_stochatreat_output_no_null_treats(treatments_dict):
...
+        data=data_with_random_index,
+        block_cols="block",
+        probs=probs,
+        treats=2,
+        idx_col=idx_col,
+        misfit_strategy=misfit_strategy,
+    )
+
+    # align the idx_col of each with the index of the input data
+    idx_cols = pd.concat(
```
Comment (RoyalTS) [inline tests/test_stochatreat_input_output.py]: Would just sort both input and output by (pandas) index and check if the index columns are then equal without any additional sorting.

Comment (redouane-dziri) [reply tests/test_stochatreat_input_output.py]: yup done

**Comment by redouane-dziri** [comment]
reformatted the docstrings here to respect line length too so that the changes are merged with the new tests

**Comment by RoyalTS** [comment]
fixes #14

**Pull request merged**